test/internal/request/complement.spec.js

```diff
@@ -0,0 +1,113 @@
+var expect = require("chai").expect;
+var complement = require("../../../lib/request/complement");
+var findPartialIntersections = require("../../../lib/request/complement").findPartialIntersections;
+
+describe("complement", function() {
+    it("returns null if no paths are provided", function() {
+        expect(complement([], [], {})).to.be.null;
+    });
+
+    it("returns null if no deduping was possible", function() {
+        var paths = [["videos", 0, "title"]];
+        expect(complement(paths, paths, {})).to.be.null;
+    });
+
+    it("returns the complement and intersection consisting of paths that can be partially deduped", function() {
```

Comment:
Reminder to myself to ask in person what's going on in this test. 😄

Reply:
For posterity: `complement()` calculates which of the paths are contained in the passed-in tree, which is a representation of paths already being requested. 'Intersection' and 'complement' refer to paths that do and do not, respectively, appear in the tree.